src/server/main_service.cc

```diff
@@ -1277,12 +1283,15 @@ void Service::DispatchManyCommands(absl::Span<CmdArgList> args_list,
 
     // Dispatch non squashed command only after all squshed commands were executed and replied
     DispatchCommand(args, cntx);
```

Comment:
We are calling DispatchCommand which in case of puase will make us wait. It could be that between the check above of IsPaused and this line the SetPause was called. So it will be safer to breake here incase IsPaused is true. I understand this flow is not triggered now as we call DispatchCommand here only for multi/eval, but if for some reason we will change the flow and call DispatchCommand in other cases this will create a bug.